### DISCUSSION TITLE
Access to config title and tagline in docs md/mdx?

### DISCUSSION THREAD

**Discussion by pke** [facebook/docusaurus#8325] [Q&A] [answered]
Do we have a way to access config values like title and tagline in our docs markdown/mdx files somehow?

**Comment by slorber** [comment, marked as answer]
You can create a React component that accesses those values in `useDocusaurusContext()` and use the component in MDX.

You could as well create a remark variable substitution plugin that replaces all {siteTitle} by the actual value. (it might already exist in the community)

An alternative that also works is to use this import `import siteConfig from '@generated/docusaurus.config';`
(this import is not something I would recommend because it's undocumented internal, but stable for years so...)

I understand that this is not an awesome DX and we'll try to solve this problem more officially later. I'm not 100% sure what is the best approach yet.

**Reply by pke** [reply]
Thanks @slorber for your comprehensive answer. I'll go the MDX way until an official way is announced (or I propose one via PR ;))

**edit**: MDX Variant works great!

**Reply by slorber** [reply]
Just tested, and this also works, because MDX components are regular components and do not usually contain any conditional logic (in most cases):

```mdx
# My Doc Title

import useDocusaurusContext from "@docusaurus/useDocusaurusContext"

Docusaurus site tagline: {useDocusaurusContext().siteConfig.tagline} 
```

![CleanShot 2022-11-18 at 13 03 44](https://user-images.githubusercontent.com/749374/202701152-8c5f4c13-16fb-42cb-88ea-d89df972f85f.png)

**Reply by pke** [reply]
Thats even better and maybe a hint how it could be solved "officially"

**edit**: It does just render the `{useDocusaurusContext().siteConfig.tagline}` into the document here on docusaurus 2.2.0

**Reply by pke** [reply]
@slorber any idea why it does not render the tagline but the code fragment instead?

**Reply by slorber** [reply]
Hmmm

I may have tried this on my MDX 2.0 branch 😅 it should work soon

In the meantime you may need to wrap it into a DOM element like `<span>` or something

**Reply by pke** [reply]
works with the `<span>` wrapped around. Magic!

Thanks!
